### DISCUSSION TITLE
ORD-SERVER and ORD-CLIENT NEW FEATURES HELP DECLUTTER THE BLOCKCHAIN WITH A PREVIEW PROCESS FOR NEW NFTS PSUEDOCODE BASIC IDEA DEVELOPERS OF ORD PLEASE READ

### DISCUSSION THREAD

**Discussion by c4pt000** [ordinals/ord#2890] [Ideas]
# ORD SERVER SETUP

ORD-SERVER DECLUTTER THE BLOCKCHAIN WITH A PREVIEW PROCESS AND ORD-SERVER BECOMES DECENTRALIZED SO THAT
 EACH ORD-SERVER NFT RECEIVING ADDRESS IS UNIQUE
THIS WAY EACH PERSON RUNNING ORD-SERVER AS A WEBADMIN DOES THEIR PART TO HELP DECLUTTER THE BITCOIN BLOCKCHAIN

    Installing Bitcoin Core
    Syncing the Bitcoin blockchain with txindex=1
    Creating a Bitcoin Core wallet 
    make sure Bitcoin core wallet is running and ord can connect
    ord wallet ord-server-addr create
    ord wallet ord-server-addr receive

    at this point the server DOES NOT HAVE TO fund their receive address
    since it will be receving transactions for people that wish to inscribe NFTs
    to that particular ord-server instance
  
  
    # how the ord wallet server address could work with an ord-server admin panel
    
    ord wallet ord-server-addr receive <-- this returns a receving address that is manually    
    copied and pasted into http://ordinals.com/admin
    under the server receiving address so that users that want to inscribe NFTs go through
    a preview process where each NFT is manually approved THIS ADDRESS BECOMES PUBLICLY VISIBLE TO ANYONE
    THAT WANTS TO INSCRIBE THEIR NFT 
    ON THAT PARTICULAR ORD-SERVER INSTANCE at the top of https://ordinals.com in a floating banner
    each person running a different copy of ORD-SERVER can have a different receiving preview address
   (almost like ord-server itself becomes decentralized with preview mode
    to filter the blockchain from clutter and obscene NFTs that take up a waste of space)
    if a user tries to inscribe an NFT and its not approved on someone's copy of ord-server then the user can try
   to go through the preview process on a different copy of ord-server
    run by someone different using another unique preview address setup on that particular ord-server
   (JUST AS LOOSE BASED EXAMPLE OF HOW THIS MIGHT WORK TO DECLUTTER THE BLOCKCHAIN)
    
    when a client sends the preview NFT to the ord-server-address every instance of ord-server 
    that sets this up will be unique and each administrator running ord-server as a web-service
    will have the benefits of the address going to their server through the blockchain
    
    if after a maximum of 14 days the NFT preview process will expire if not approved will return
    the inscribed fees and amount back to the sender address that it was received from

    if the NFT preview process for any NFT is approved in under 14 days maximum time then the NFT
    will inscribe and defintiely commit to the blockchain and will be under the public view of inscriptions
    or collections
    
    new features for ordianls.com when a user is running ord as "ord server"
    only the web admin running "ord server" can access the admin panel and the preview panel using rpcuser
    and rpcpassword from bitcoin.conf
    /admin panel
    /preview panel

  
   # ORD CLIENT SIDE TO SEND AND RECEIVE NFTS

    Installing Bitcoin Core
    Syncing the Bitcoin blockchain with txindex=1
    Creating a Bitcoin Core wallet
   
    ord wallet create
    ord wallet receive
    at this point the client funds their receive address to inscribe NFTs 

    now instead of sending the NFT to the blockchain like this!
    OLDWAY WITH TOO MUCH BLOCKCHAIN_CLUTTER-----> ord wallet inscribe --fee-rate FEE_RATE --file FILE
    
    
    
    users send the NFT to the blockchain specific address like this created on the ord server side
    NEWWAY WITH NFT PREVIEW MODE ENABLED ON ORD SERVER------> ord wallet inscribe --ord-server-addr <ord-server-receive-address-here> --fee-rate FEE_RATE --file FILE
    
    once the users inscribes the NFT to the nft-server receiving address the NFT goes into preview mode
    for a maximum of up to 14 days. IF AFTER 14 days the NFT is not approved by the webadmin or moderator of ord-server
    the received inscription is returned back to the user that sent it and it is not finalized into the blockchain
    
    if the NFT preview process for any NFT is approved in under 14 days maximum time then the NFT
    will inscribe and defintiely commit to the blockchain and will be under the public view of inscriptions
    or collections

**Comment by cliobitcoinbank** [comment]
Yes and you can use regtest or use a public regtest node like the one hosted by https://hash.art/ to test what your ordinals will look and behave like when inscribed to prevent any errors pr waste. Regtest is a 1:1 equivalent to bitcoin mainnet and used as a testing environment when privacy is needed. Ordinals devs have this tooling available although it is still underutilized imo.
